modin/engines/base/io.py

```diff
@@ -310,36 +311,38 @@ def read_excel(
         if skip_footer != 0:
             skipfooter = skip_footer
         ErrorMessage.default_to_pandas("`read_excel`")
-        return cls.from_pandas(
-            pandas.read_excel(
-                io,
-                sheet_name=sheet_name,
-                header=header,
-                names=names,
-                index_col=index_col,
-                parse_cols=parse_cols,
-                usecols=usecols,
-                squeeze=squeeze,
-                dtype=dtype,
-                engine=engine,
-                converters=converters,
-                true_values=true_values,
-                false_values=false_values,
-                skiprows=skiprows,
-                nrows=nrows,
-                na_values=na_values,
-                keep_default_na=keep_default_na,
-                verbose=verbose,
-                parse_dates=parse_dates,
-                date_parser=date_parser,
-                thousands=thousands,
-                comment=comment,
-                skipfooter=skipfooter,
-                convert_float=convert_float,
-                mangle_dupe_cols=mangle_dupe_cols,
-                **kwds
-            )
+        parsed = pandas.read_excel(
+            io,
+            sheet_name=sheet_name,
+            header=header,
+            names=names,
+            index_col=index_col,
+            parse_cols=parse_cols,
+            usecols=usecols,
+            squeeze=squeeze,
+            dtype=dtype,
+            engine=engine,
+            converters=converters,
+            true_values=true_values,
+            false_values=false_values,
+            skiprows=skiprows,
+            nrows=nrows,
+            na_values=na_values,
+            keep_default_na=keep_default_na,
+            verbose=verbose,
+            parse_dates=parse_dates,
+            date_parser=date_parser,
+            thousands=thousands,
+            comment=comment,
+            skipfooter=skipfooter,
+            convert_float=convert_float,
+            mangle_dupe_cols=mangle_dupe_cols,
+            **kwds
         )
+        if isinstance(parsed, OrderedDict):
+            return parsed
```

Comment:
This should convert each of the `pandas.DataFrame` into `modin.pandas.DataFrame`. You can use `cls.from_pandas` to do it.